Cp 14449 add tests

roberthocking opened the pull request:
# Description of the issue
Adding smoke tests

# Description of changes
added small smoke test to check for exsistence of log groups

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
ran test on cloud9 instance

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

roberthocking committed:
Updates to make file and added tests

updated the create-eks-cluster to use a configfile
add config file for clsuter creation
added integration test skeleton

roberthocking committed:
fixed typo

dmepham reviewed (commented):
the structure looks good to me! the only comment I might have is that I wonder if test result logs might get lost if we run the `make destroy-eks-cluster` right after. IIRC the eksctl command is pretty verbose. maybe it would be clearer if test setup/teardown was separate from test execution? but I think either way works

roberthocking committed:
improvements

updated Make file based on comments from review

roberthocking committed:
moved tests to be in their own package

roberthocking committed:
Moved tests into proper locations
Added integration test to check if log group is properly created for new cluster
Fixed tons of typos in other file

roberthocking committed:
Updates for Integration tests

Updated make command to properly run integration and/or smoke tests
Added integration test to cloudwatchlogs package
fixed typo in NodeGroups.yaml

dmepham commented on Makefile:
```diff
@@ -309,3 +308,9 @@ init:
 	fi
+
+smoke-test:
+	go test -short -coverprofile smoke_coverage.txt -failfast ./awscsm/... ./cfg/... ./cmd/... ./handlers/... ./internal/... ./logger/... ./logs/... ./metric/... ./plugins/... ./profiler/... ./tool/... ./translator/...
```
`smoke_coverage.txt` is generated on every run, right? is it worth adding to the .gitignore?

roberthocking replied:
Good idea. I'll add it

dmepham reviewed (approved):
lgtm! just one small comment. it might be nice to have a section in the README for our testing, but I'm not sure how that would work in a forked project with a bunch of non-CZ stuff in the README already.

roberthocking committed:
updated readme with instruction on how to run tests
